### PR TITLE
Removed 'record_solver_output' recording option and moved 'record_derivatives' from system to driver

### PR DESCRIPTION
**Pull request by ryanfarr01**

### CODE REVIEW

File: openmdao/recorders/tests/test_sqlite_recorder.py [hunk 1]
```diff
@@ -643,7 +642,15 @@ def test_record_line_search_armijo_goldstein(self):
         coordinate = [0, 'Driver', (0,), 'root._solve_nonlinear', (0,), 'NewtonSolver', (3,), 'ArmijoGoldsteinLS', (4,)]
         expected_abs_error = 3.49773898733e-9
         expected_rel_error = expected_abs_error / 2.9086436370499857e-08
-        expected_solver_output = None
+        expected_solver_output = {
+            "con_cmp1.con1": [-22.42830237000701],
```
Comment (swryan) [inline openmdao/recorders/tests/test_sqlite_recorder.py]: that's a lot of precision in these tests, the numbers don't vary at all across platforms?

File: openmdao/solvers/solver.py [hunk 2]
```diff
@@ -248,14 +246,13 @@ def _setup_solvers(self, system, depth):
...
-            else:  # it's a LinearSolver
-                outputs = self._system._vectors['output']['linear']
-            # myoutputs = {n for n in outputs
-            myoutputs = {n for n in outputs._names
-                         if check_path(n, incl, excl)}
+        if isinstance(self, NonlinearSolver):
+            outputs = self._system._outputs
+        else:  # it's a LinearSolver
+            outputs = self._system._vectors['output']['linear']
+        # myoutputs = {n for n in outputs
```
Comment (swryan) [inline openmdao/solvers/solver.py]: could clean up this commented code

File: openmdao/recorders/tests/test_sqlite_recorder.py (hunk 1 above)
Comment (ryanfarr01) [reply openmdao/recorders/tests/test_sqlite_recorder.py]: Oh I'm sure the numbers do vary across platforms, though the assertions on these tests check that the value is in an expected range defined by the tolerance `self.eps`

Comment (swryan) [reply openmdao/recorders/tests/test_sqlite_recorder.py]: ah right, of course...  just seemed a little odd to see all those digits :)

**Pull request merged**